### DISCUSSION TITLE
[IDEA] FaaS and RR

### DISCUSSION THREAD

**Discussion by sash2222** [roadrunner-server/roadrunner#781] [Ideas]
Guys, I just have an idea that is actively spinning in my head for discussion. Maybe it doesn't make any sense.
What if using the local FaaS platform (functions as a service) using the example of Fission (a very interesting project), run PHP functions in RR via API? In a sense, PP Jobs performs this function, but not quite so. As I see it, there are a number of RR servers, "empty" with no code, in the form of a runtime environment. Is there a Fission that somehow manages jobs (reads from a queue or something else) and how does it receive a job, runs a piece of PHP (function) in RR? It may be easy to cram PHP into the runtime of PHP for Fission, RR, but something tells me that it is unlikely to work just like that.

**Comment by wolfy-j** [comment]
Just FYI: https://github.com/spiral/roadrunner-docs/blob/1.0/library/aws-lambda.md

You can use the worker pool pretty much directly. So this is def possible.

**Comment by rustatian** [comment]
RR2: https://github.com/spiral/roadrunner-docs/blob/master/library/aws-lambda.md
